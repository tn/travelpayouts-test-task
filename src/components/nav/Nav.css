.nav {
    margin-top: 44px;
}

.nav__list {
    display: flex;
    flex-flow: column;
    margin: 0;
    padding: 0;
    list-style-type: none;
}

.nav__item {
    display: block;
}

.nav__link {
    display: inherit;
    font-size: 0;
    width: 100%;
    height: 72px;
    background: url('./circle.svg') center center no-repeat;
    text-decoration: none;
    opacity: 0.5;
}

.nav__link--active,
.nav__link:hover {
    opacity: 1;
}
