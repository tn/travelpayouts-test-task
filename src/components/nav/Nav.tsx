import * as React from 'react'
import './Nav.css'

const activeIndex = 6
const itemsArr = new Array(8).fill(1)

const Nav: React.SFC = () => (
  <nav className='nav'>
    <ul className='nav__list'>
      {itemsArr.map((item, index) => (
        <li key={`item${index}`} className='nav__item'>
          <a className={`nav__link ${index === activeIndex ? 'nav__link--active' : ''}`} href='#' />
        </li>
      ))}
    </ul>
  </nav>
)

export default Nav
