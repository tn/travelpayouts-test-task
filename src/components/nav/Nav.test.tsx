import * as React from 'react'
import * as Adapter from 'enzyme-adapter-react-16'
import { configure, shallow } from 'enzyme'
import * as renderer from 'react-test-renderer'
import Nav from './Nav'

configure({ adapter: new Adapter() })

describe('Nav', () => {
  it('should be rendered', () => {
    const wrapper = renderer.create(<Nav />).toJSON()

    expect(wrapper).toMatchSnapshot()
  })

  it('should have active link', () => {
    const wrapper = shallow(<Nav />)

    expect(wrapper.find('.nav__link--active').exists()).toBeTruthy()
  })
})
