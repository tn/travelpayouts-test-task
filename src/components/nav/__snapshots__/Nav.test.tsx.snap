// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should be rendered 1`] = `
<nav
  className="nav"
>
  <ul
    className="nav__list"
  >
    <li
      className="nav__item"
    >
      <a
        className="nav__link "
        href="#"
      />
    </li>
    <li
      className="nav__item"
    >
      <a
        className="nav__link "
        href="#"
      />
    </li>
    <li
      className="nav__item"
    >
      <a
        className="nav__link "
        href="#"
      />
    </li>
    <li
      className="nav__item"
    >
      <a
        className="nav__link "
        href="#"
      />
    </li>
    <li
      className="nav__item"
    >
      <a
        className="nav__link "
        href="#"
      />
    </li>
    <li
      className="nav__item"
    >
      <a
        className="nav__link "
        href="#"
      />
    </li>
    <li
      className="nav__item"
    >
      <a
        className="nav__link nav__link--active"
        href="#"
      />
    </li>
    <li
      className="nav__item"
    >
      <a
        className="nav__link "
        href="#"
      />
    </li>
  </ul>
</nav>
`;
